internal/graph/resolve_check_response.go

```diff
@@ -0,0 +1,51 @@
+package graph
+
+// CloneResolveCheckResponse clones the provided ResolveCheckResponse.
+//
+// If 'r' defines a nil ResolutionMetadata then this function returns
+// an empty value struct for the resolution metadata instead of nil.
+func CloneResolveCheckResponse(r *ResolveCheckResponse) *ResolveCheckResponse {
```

Comment:
```suggestion
func (r *ResolveCheckResponse) Clone() *ResolveCheckResponse {
```

this makes the function discoverable when using auto-complete :) also applies to the request object.